code/game/objects/items/flag.dm

```diff
@@ -267,8 +268,16 @@
 
 /obj/item/flag/chameleon/burn()
 	if(boobytrap)
-		boobytrap.prime()
-	..()
+		fire_act()
+		addtimer(CALLBACK(src, .proc/prime_boobytrap), boobytrap.det_time)
+	else
+		..()
+
+/obj/item/flag/chameleon/proc/prime_boobytrap()
+	boobytrap.forceMove(get_turf(loc))
+	boobytrap.prime()
+	boobytrap = null
+	burn()
```

Comment:
Won't this now cause a delayed burn? You use the lighter and then after `det_time` the flag actually starts burning visually

Reply:
The lighter calls burn(), which calls fire_act() before the timer. Fire_act causes the flag to burn visually. The timer then calls burn() again after the explosive does it's thing to prevent the flag from burning forever.

Reply:
You're right. All good here